File: Project/SpringBoot/src/main/java/com/catcare/project/Service/TratamientoService.java
package com.catcare.project.Service;

import java.util.Collection;

import com.catcare.project.Entity.Tratamiento;

public interface TratamientoService {
    
    // Obtener todos los tratamientos
    public Collection<Tratamiento> SearchAll();

    // Buscar un tratamiento por su ID
    public Tratamiento SearchById(Long id);

    // Agregar un nuevo veterinario
    public Tratamiento add(Tratamiento tratamiento);

     // Eliminar un tratamiento por su ID
     public void deleteById(Long id);

    // Actualizar los datos de un tratamiento
    public void update(Tratamiento tratamiento);

    // Devuelve la cantidad de tratamientos administrados en el último mes
    public Long countTratamientosLastMonth();

    // Devuelve la cantidad de mascotas Activas
    public Long countMascotasEnTratamiento();


}

File: Project/Angular/src/app/service/tratamiento/tratamiento.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class TratamientoService {

  private ROOT_URL = 'http://localhost:8090/catcare/tratamientos';

  constructor(private http: HttpClient) {}

  // Obtener todos los tratamientos
  getAllTratamientos(): Observable<any[]> {
    return this.http.get<any[]>(`${this.ROOT_URL}/all`);
  }

  // Obtener un tratamiento por su ID
  getTratamientoById(tratamientoId: number): Observable<any> {
    return this.http.get<any>(`${this.ROOT_URL}/find?id=${tratamientoId}`);
  }

  // Agregar un nuevo veterinario
  agregarTratamiento(tratamiento: any): Observable<any> {
    return this.http.post<any>(`${this.ROOT_URL}/agregar`, tratamiento);
  }

  // Eliminar un tratamiento por su ID
  eliminarTratamiento(tratamientoId: number): Observable<any> {
    return this.http.delete<any>(`${this.ROOT_URL}/delete/${tratamientoId}`);
  }

  // Actualizar los datos de un tratamiento
  actualizarTratamiento(tratamientoId: number, tratamiento: any): Observable<any> {
    return this.http.put<any>(`${this.ROOT_URL}/update/${tratamientoId}`, tratamiento);
  }
}
